packages/components/src/Combobox/hooks/useCombobox.ts

```diff
@@ -14,18 +21,44 @@ type UseComboboxReturn = {
   selectedOptions: ComboboxOption[];
   selectedStateSetter: (selection: ComboboxOption[]) => void;
   shouldScroll: MutableRefObject<boolean>;
+  internalFilteredOptions: ComboboxOption[];
+  handleSearchChange: (value: string) => void;
 } & UseMakeComboboxHandlersReturn;
 
 export function useCombobox(
   selected: ComboboxOption[],
   onSelect: (selection: ComboboxOption[]) => void,
+  options: ComboboxOption[],
   onClose?: () => void,
   multiSelect?: boolean,
+  onSearchChange?: (searchValue: string) => void,
+  debounceTime: number = 300,
 ): UseComboboxReturn {
   const wrapperRef = useRef<HTMLDivElement>(null);
   const shouldScroll = useRef<boolean>(false);
   const [open, setOpen] = useState<boolean>(false);
   const [searchValue, setSearchValue] = useState<string>("");
+  const [internalFilteredOptions, setInternalFilteredOptions] =
+    useState<ComboboxOption[]>(options);
+
+  const searchChangeCallback = useCallback(
+    debounce(
+      (value: string) => onSearchChange && onSearchChange(value),
+      debounceTime,
+    ),
+    [],
+  );
+
+  const debouncedFilterOptions = useCallback(
+    debounce((value: string) => {
+      const filtered = options.filter(option => {
+        return option.label.toLowerCase().includes(value.toLowerCase());
+      });
+
+      setInternalFilteredOptions(filtered);
+    }, debounceTime),
```

Comment:
is it really worth debouncing the internal version?

filtering a local set of data isn't very expensive, and there are certain interactions that feel laggy or unresponsive as a result, for example it'll delay a moment before updating the filtered list but since we immediately update the input value and that's tied to the "no results" message you can have a moment where it'll say "no results for ''" rather than immediately going back to the unfiltered state

of course it's possible to fix that, it just means we need to either debounce everything that happens as a result of typing in the input, or we move everything into a single debounced handler

Reply:
![Kapture 2024-01-17 at 08 36 58](https://github.com/GetJobber/atlantis/assets/11843215/c9b717af-a418-4109-860e-801d99ae32a7)

Reply:
If we can avoid doing loading (or loading-esque, which is how I think of debouncing) delays in scenarios where they're not needed, it should feel much better. _(Is `scenarios where they're not needed` hard to know?)_

Reply:
stopped debouncing the internal callback to search/filter since it only led to a feeling of unresponsiveness and it's not at all necessary